ctapipe/tools/apply_models.py

```diff
@@ -146,6 +146,9 @@ def start(self):
                 reconstructor,
             )
             self._combine(reconstructor.stereo_combiner, mono_predictions)
+            self.h5file.close()
```

Comment:
You might want to add a comment here to remind a future developer why this close/open was added, otherwise it may get removed.

Reply:
I am not really happy yet, there should be a better solution...